internal/controllers/topology/cluster/reconcile_state.go

```diff
@@ -676,6 +787,9 @@ func (r *Reconciler) updateMachineDeployment(ctx context.Context, s *scope.Scope
 
 	log.Infof("Patching %s", tlog.KObj{Obj: currentMD.Object})
 	if err := patchHelper.Patch(ctx); err != nil {
+		// Best effort cleanup of the InfrastructureMachineTemplate & BootstrapTemplate (only on creation).
```

Comment:
```suggestion
		// Best effort cleanup of the InfrastructureMachineTemplate & BootstrapTemplate (only on template rotation).
```
(same as l.778 I think)